avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -417,32 +437,73 @@ int getSchemaId() {
       return schemaId;
     }
 
-    Object read(Schema writerSchema) {
-      return read(writerSchema, specificAvroReaderSchema);
+    Object read(AvroSchema writerAvroSchema) {
+      return read(writerAvroSchema,
+          specificAvroReaderSchema != null ? new AvroSchema(specificAvroReaderSchema) : null);
     }
 
-    Object read(Schema writerSchema, Schema readerSchema) {
+    Object read(AvroSchema writerAvroSchema, AvroSchema readerAvroSchema) {
```

Comment:
It would be good to document the steps/order of applying the rules.

Reply:
Added some inline documentation